@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1756,10 +1780,9 @@ export class TextElementsRenderer {
             this.m_viewState.env,
             this.m_tmpVector3
         );
-        // Only process labels frustum-clipped labels
-        if (
-            this.m_screenProjector.projectOnScreen(worldPosition, tempScreenPosition) === undefined
-        ) {
+
+        // Only process labels that are potentially within the frustum.
+        if (!this.labelPotentiallyVisible(worldPosition, tempScreenPosition)) {
             return false;
```

Comment:
This is too soon to do check label bound visibility. The label has not been yet placed, therefore you don't even know the alignment wrt the anchor point. You're assuming the label is centered but it could be left-aligned, and then it would be skipped even being fully inside the screen.

Here we can only check if it's within the far distance range. The other checks have to be done after placement.

Reply:
Again, I misunderstood. This should be fine.